test/cmdline_test.py

```diff
@@ -202,21 +202,14 @@ def test_bin_luigi_help_class(self):
 class NewStyleParameters822Test(unittest.TestCase):
     # See https://github.com/spotify/luigi/issues/822
 
+    @parsing(['FooSubClass', '--x', 'xyz', '--FooBaseClass-x', 'xyz'])
     def test_subclasses(self):
-        ap = luigi.interface._ArgParseInterface()
-
-        task, = ap.parse(['--local-scheduler', '--no-lock', 'FooSubClass', '--x', 'xyz', '--FooBaseClass-x', 'xyz'])
-        self.assertEquals(task.x, 'xyz')
-
-        # This won't work because --FooSubClass-x doesn't exist
-        self.assertRaises(BaseException, ap.parse, (['--local-scheduler', '--no-lock', 'FooBaseClass', '--x', 'xyz', '--FooSubClass-x', 'xyz']))
```

Comment:
I think you wrote this @erikbern. I had to remove this `assertRaises` as I did not understand it. It doesn't make sense to me at first sight. Can you comment?